### PR TITLE
Breaking: Drop `codeframe` and `table` formatters

### PR DESCRIPTION
**Pull request by fregante**
Fixes https://github.com/eslint/eslint/issues/14277
Related to https://github.com/eslint/eslint/issues/14098

#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/master/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[ ] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-change-proposal.md))
[ ] Add autofixing to a rule
[ ] Add a CLI option
[ ] Add something to the core
[x] Other, please explain:

Removal of features. Breaking

#### What changes did you make? (Give an overview)

I dropped the `table` and `codeframe` formatters from the core as discussed in https://github.com/eslint/eslint/issues/14277

### CODE REVIEW

File: lib/cli-engine/cli-engine.js [hunk 1]
```diff
@@ -1018,7 +1018,11 @@ class CLIEngine {
             try {
                 return require(formatterPath);
             } catch (ex) {
-                ex.message = `There was a problem loading formatter: ${formatterPath}\nError: ${ex.message}`;
+                if (format === "table" || format === "codeframe") {
+                    ex.message = `The ${format} formatter is no longer part of core ESLint. Install it manually with \`npm install -D ${format}\``;
```
Comment (TimvdLippe) [inline lib/cli-engine/cli-engine.js:1022]: I think the install command should be `npm install -D eslint-formatter-${format}` per https://eslint.org/docs/user-guide/command-line-interface#-f-format, should it not?

**Comment by bl-ue** [comment]
ping @fregante — any progress?  Would be cool to get this merged 😎

**Comment by fregante** [comment]
Sorry for the delay!

File: tests/lib/cli-engine/cli-engine.js [hunk 2]
```diff
@@ -4647,6 +4647,18 @@ describe("CLIEngine", () => {
             }, `There was a problem loading formatter: ${fullFormatterPath}\nError: Cannot find module '${fullFormatterPath}'`);
         });
 
+        it("should return null when a built-in formatter no longer exists", () => {
```
Comment (lydell) [inline tests/lib/cli-engine/cli-engine.js:4650]:
```suggestion
        it("should throw an error when a built-in formatter no longer exists", () => {
```

?

Comment (bl-ue) [reply tests/lib/cli-engine/cli-engine.js:4650]: Based on the other cases, I'd say `throw` rather than `throw an error`, @lydell.

**Review by mdjermanovic — CHANGES_REQUESTED** [review]
@fregante can you fix https://github.com/eslint/eslint/pull/14316#discussion_r645127722, and rebase (there's a conflict in `package.json`)?

Everything else LGTM

File: tests/lib/cli-engine/cli-engine.js (hunk 2 above)
Comment (fregante) [reply tests/lib/cli-engine/cli-engine.js:4650]:
```suggestion
        it("should throw when a built-in formatter no longer exists", () => {
```

**Review by mdjermanovic — APPROVED** [review]
LGTM, thanks!

**Review by snitin315 — APPROVED** [review]
LGTM

**Review by nzakas — APPROVED** [review]

**Comment by TimvdLippe** [comment]
I see that the 8.0.0 milestone has been created, excited for that! Can we add this PR to the milestone as well? Thanks!

**Comment by nzakas** [comment]
Only issues are in v8.0.0 board. This PR is there by virtue of the linked issue.

**Comment by TimvdLippe** [comment]
Ah of course. Sorry for the noise!

**Comment by fregante** [comment]
If anyone wants to maintain these 2 formatters, let me know, they're available as standalone projects:

- https://github.com/fregante/eslint-formatter-codeframe
- https://github.com/fregante/eslint-formatter-table

**Pull request merged**